packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -433,133 +618,233 @@ fn array_to_user_warnings(array: mv8::Value<'_>) -> Vec<UserWarning> {
 }
 
-impl<'m> RunnerImpl<'m> {
-    fn load_datasets(mv8: &'m MiniV8, shared_ctx: &SharedStore) -> Result<mv8::Value<'m>> {
-        let js_datasets = mv8.create_object();
+impl<'s> ThreadLocalRunner<'s> {
+    fn load_datasets(
+        scope: &mut v8::HandleScope<'s>,
+        shared_ctx: &SharedStore,
+    ) -> Result<v8::Local<'s, v8::Value>> {
+        let js_datasets = v8::Object::new(scope);
         for (dataset_name, dataset) in shared_ctx.datasets.iter() {
-            let js_name = mv8.create_string(dataset_name.as_str());
+            let js_name = new_js_string(scope, &dataset_name)?;
 
             let json = dataset.data();
             // TODO: Use `from_utf8_unchecked` instead here?
             //       (Since datasets' json can be quite large.)
             let json =
                 std::str::from_utf8(json).map_err(|_| Error::Unique("Dataset not utf8".into()))?;
-            let json = mv8.create_string(json);
+            let json = new_js_string(scope, json)?;
 
-            js_datasets.set(js_name, json)?;
+            js_datasets
+                .set(scope, js_name.into(), json.into())
+                .ok_or_else(|| Error::V8("Could not set property on Object".to_string()))?;
         }
-        Ok(mv8::Value::Object(js_datasets))
+
+        Ok(js_datasets.into())
     }
 
-    pub fn new(mv8: &'m MiniV8, init: &ExperimentInitRunnerMsg) -> Result<Self> {
-        let embedded = Embedded::import(mv8)?;
-        let datasets = Self::load_datasets(mv8, &init.shared_context)?;
+    pub fn new(
+        scope: &mut v8::HandleScope<'s>,
+        context: v8::Local<'s, v8::Context>,
+        init: &ExperimentInitRunnerMsg,
+    ) -> Result<Self> {
+        let embedded = Embedded::import(scope, context)?;
+        let datasets = Self::load_datasets(scope, &init.shared_context)?;
 
-        let pkg_fns = mv8.create_array();
-        let pkg_init_msgs = mv8.create_array();
         let pkg_config = &init.package_config.0;
+        let pkg_fns = v8::Array::new(scope, pkg_config.len() as i32);
+        let pkg_init_msgs = v8::Array::new(scope, pkg_config.len() as i32);
         for (i_pkg, pkg_init) in pkg_config.values().enumerate() {
             let i_pkg = i_pkg as u32;
 
-            let pkg_name = format!("{}", &pkg_init.name);
-            let pkg = JsPackage::import(mv8, &embedded, &pkg_name, pkg_init.r#type)?;
+            let pkg_name = pkg_init.name.to_string();
+            let pkg = JsPackage::import(scope, context, &embedded, &pkg_name, pkg_init.r#type)?;
             tracing::trace!(
                 "pkg experiment init name {:?}, type {}, fns {:?}",
                 &pkg_init.name,
                 &pkg_init.r#type.as_str(),
                 &pkg.fns,
             );
-            pkg_fns.set(i_pkg, pkg.fns)?;
+            pkg_fns
+                .set_index(scope, i_pkg, pkg.fns.into())
+                .ok_or_else(|| {
+                    Error::V8(format!("Could not set index {i_pkg} on pkg_fns array"))
```

Comment:
Same comment as all of the ones above about could not set index

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/2a832f8dc26357b06320aed8e66cb59822d849a2